### PR TITLE
Add support for resources.

### PR DESCRIPTION
**Pull request by ericsnowcurrently**

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/bundlechanges/52/
Test FAILed.

File: changes.go [hunk 1]
```diff
@@ -223,7 +223,11 @@ func (ch *AddServiceChange) GUIArgs() []interface{} {
 	if endpointBindings == nil {
 		endpointBindings = make(map[string]string, 0)
 	}
-	return []interface{}{ch.Params.Charm, ch.Params.Service, options, ch.Params.Constraints, storage, endpointBindings}
+	resources := ch.Params.Resources
+	if resources == nil {
+		resources = make(map[string]int, 0)
+	}
+	return []interface{}{ch.Params.Charm, ch.Params.Service, options, ch.Params.Constraints, storage, endpointBindings, resources}
```
Comment (frankban) [inline changes.go]: The GUI should be ready to accept the new resources argument before adding them here.
I'd say let's leave GUIArgs untouched for now.

Comment (frankban) [inline changes.go]: Maybe add a TODO comment?

**Comment by frankban** [comment]
:+1: with a change.
Thank you!

File: changes.go (hunk 1 above)
Comment (ericsnowcurrently) [inline changes.go]: Sounds good.  I'm glad you noticed this.  It was the one thing I was unsure about. :)

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/bundlechanges/53/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/bundlechanges/54/
Test PASSed.

**Comment by natefinch** [comment]
:+1:

**Comment by kat-co** [comment]
+1

**Comment by ericsnowcurrently** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/bundlechanges-merge

**Pull request merged**